xgi/drawing/layout.py

```diff
@@ -337,3 +346,159 @@ def pca_transform(pos, theta=0, degrees=True):
+
+    Returns
+    -------
+    pos : dict
+        A dictionary of positions keyed by node
+    """
+    if center is None:
+        center = [0,0]
+    
+    if H.num_nodes == 0:
+        pos = {}
+        return pos
+    elif H.num_nodes == 1:
+        pos = {list(H.nodes)[0]: center}
+        return pos
+
+    pos = []
+    if equidistant:
+        chord = 1
+        step = 0.5
+        theta = resolution
+        theta += chord / (step * theta)
+        for _ in range(len(H)):
+            r = step * theta
+            theta += chord / r
+            pos.append([np.cos(theta)*r+center[0], np.sin(theta)*r+center[1]])
+    else:
+        dist = np.arange(len(H), dtype=float)
+        angle = resolution * dist
+        pos = np.transpose(dist * np.array([np.cos(angle), np.sin(angle)]))
+    
+    pos = dict(zip(list(H.nodes), pos))
+    
+    return pos
+    
+
+def barycenter_kamada_kawai_layout(H, return_phantom_graph=False):
+    """
+    Position nodes using Kamada-Kawai path-length cost-function
+    using an augmented version of the the graph projection
+    of the hypergraph (or simplicial complex), where phantom nodes
+    (barycenters) are created for each edge composed by more than two nodes.
+    If a simplicial complex is provided the results will be based on the
+    hypergraph constructed from its maximal simplices.
+
+    Parameters
+    ----------
+    H : xgi Hypergraph or SimplicialComplex
+        A position will be assigned to every node in H.
+    return_phantom_graph: bool (default=False)
+        If True the function returns also the augmented version of the 
+        the graph projection of the hypergraph (or simplicial complex).
+
+    Returns
+    -------
+    pos : dict
+        A dictionary of positions keyed by node
+    """
+    if isinstance(H, SimplicialComplex):
+        H = convert.from_max_simplices(H)    
+    # Creating the projected networkx Graph, I will fill it manually
+    G = nx.Graph()
+
+    # Adding real nodes
+    G.add_nodes_from(list(H.nodes))
+
+    # Adding links (edges composed by two nodes only, for which we don't use phantom nodes
+    for i, j in H.edges.filterby("order", 1).members():
+        G.add_edge(i, j)
+
+    # Adding phantom nodes and connections therein
+    # I will start from the first int node-label available
+    try:
+        phantom_node_id = max([n for n in H.nodes if isinstance(n, int)]) + 1
+    except ValueError:
+        # The list of node-labels has no integers, so I start from 0
+        phantom_node_id = 0
+
+    # Looping over the hyperedges of different order (from triples up)
+    for d in range(2, max_edge_order(H) + 1):
+        # Hyperedges of order d (d=2: triplets, etc.)
+        for he in H.edges.filterby("order", d).members():
+            # Adding one phantom node for each hyperedge and linking it to the nodes of the hyperedge
+            for n in he:
+                G.add_edge(phantom_node_id, n)
+            phantom_node_id += 1
```

Comment:
This code is repeated from the other barycenter layouts, right? 
Could you extract if into a helper function, say `_add_phantom_nodes()`?

It's good practice in itself but I think it might also solve the coverage thing.